pkg/alertmanager/operator.go

```diff
@@ -935,6 +936,24 @@ func (c *Operator) selectAlertManagerConfigs(ctx context.Context, am *monitoring
 	return res, nil
 }
 
+// configureHTTPConfigInStore configure the asset store for HTTPConfigs.
+func configureHTTPConfigInStore(ctx context.Context, store *assetStore, httpConfig *monitoringv1alpha1.HTTPConfig, amcNamespace string, receiverKey string) {
```

Comment:
Let's make it a method of `assetStore`. And it should return an `error`.

```suggestion
func (a *assetStore) configureHTTPConfigInStore(ctx context.Context, httpConfig *monitoringv1alpha1.HTTPConfig, namespace string, key string) error {
```

Reply:
Thanks that looks way better